src/libcollections/str.rs

```diff
@@ -1055,8 +1055,8 @@ impl str {
     /// An iterator over substrings of the given string slice, separated by a
     /// pattern, restricted to returning at most `count` items.
     ///
-    /// The last element returned, if any, will contain the remainder of the
```

Comment:
I think, as a whole, your changes are an improvement and make it easier to understand this concept.

> If a `count`th substring

I personally stumbled on this though. It took me a few re-reads to understand this is talking about "substring with `count`th number of items". I can't think of a better suggestion at the moment without making it too verbose. Going to think about this...